### PR TITLE
Add automatic releases

### PR DESCRIPTION
**Pull request by mitchellhamilton**
This only does the create a PR bit, not auto-publishing.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+          version: 10.x
+
+      - name: Install Bolt
+        run: npm install --global bolt
+
+      - name: Install Dependencies
+        run: bolt
+
+      - name: Create Release Pull Request
+        uses: changesets/action@master
```
Comment (timleslie) [inline .github/workflows/release.yml]: Is this a pre-canned action which already exists somewhere else?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - master
```
Comment (timleslie) [inline .github/workflows/release.yml]: I wonder if we *always* want to do this on push to master? A lot of the time I'd prefer to group a whole bunch of PRs in a single combined release.

File: .github/workflows/release.yml (hunk 1 above)
Comment (mitchellhamilton) [reply .github/workflows/release.yml]: Yes, I wrote it, https://github.com/changesets/action

Comment (timleslie) [reply .github/workflows/release.yml]: Aha, Excellent!

File: .github/workflows/release.yml (hunk 2 above)
Comment (mitchellhamilton) [reply .github/workflows/release.yml]: It always runs on master but that doesn't mean it creates a release on every push. It creates a PR like the ones we have now so you can do exactly that, if there are new changesets on master, the PR will be updated.

Comment (timleslie) [reply .github/workflows/release.yml]: Ah, of course, very good! Let's ship it then :-)

**Pull request merged**